metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py

```diff
@@ -88,72 +102,40 @@ def assigned_metastore(self) -> Metastore:
-
-        if num_catalogs == 0:
-            logger.info(
-                f"Catalogs not found for metastore where metastore_id is {metastore.metastore_id}"
-            )
+        for catalog in response:
+            yield self._create_catalog(metastore, catalog)
 
     def schemas(self, catalog: Catalog) -> Iterable[Schema]:
-        response: dict = self._unity_catalog_api.list_schemas(
-            catalog_name=catalog.name, name_pattern=None
-        )
-        if response.get("schemas") is None:
-            logger.info(f"Schemas not found for catalog {catalog.name}")
+        response = self._workspace_client.schemas.list(catalog_name=catalog.name)
+        if not response:
+            logger.info(f"Schemas not found for catalog {catalog.id}")
             return []
-
-        for schema in response["schemas"]:
+        for schema in response:
             yield self._create_schema(catalog, schema)
 
     def tables(self, schema: Schema) -> Iterable[Table]:
-        response: dict = self._unity_catalog_api.list_tables(
-            catalog_name=schema.catalog.name,
-            schema_name=schema.name,
-            name_pattern=None,
-        )
-
-        if response.get("tables") is None:
-            logger.info(
-                f"Tables not found for schema {schema.catalog.name}.{schema.name}"
+        with patch("databricks.sdk.service.catalog.TableInfo", TableInfoWithGeneration):
+            response = self._workspace_client.tables.list(
+                catalog_name=schema.catalog.name, schema_name=schema.name
             )
-            return []
-
-        for table in response["tables"]:
-            yield self._create_table(schema=schema, obj=table)
+            if not response:
+                logger.info(f"Tables not found for schema {schema.id}")
+                return []
+            for table in response:
+                try:
+                    yield self._create_table(schema, table)
+                except Exception as e:
+                    logger.warning(f"Error parsing table: {e}")
+                    self.report.report_warning("table-parse", str(e))
 
     def service_principals(self) -> Iterable[ServicePrincipal]:
-        # TODO: Replace with self._workspace_client.service_principals.list() when it supports pagination
-        start_index = 1  # Unfortunately 1-indexed
-        items_per_page = 0
-        total_results = float("inf")
-        while start_index + items_per_page <= total_results:
-            response: dict = self._unity_catalog_api.client.client.perform_query(
-                "GET", "/account/scim/v2/ServicePrincipals"
-            )
-            start_index = response["startIndex"]
-            items_per_page = response["itemsPerPage"]
-            total_results = response["totalResults"]
-            for principal in response["Resources"]:
-                yield self._create_service_principal(principal)
+        for principal in self._workspace_client.service_principals.list():
```

Comment:
I believe from our call, pagination is just not supported for this endpoint (even though it looks like it is). Ideally, I'd pull the list of service principals from ingestion, then call this endpoint to figure out what they are, but that would require a big code refactor and I'm not sure how we'd generate workunits in real time, rather than all at the end, if we did this. So for now, I think it's fine to just get them all at once